datepicker/operations/index.js

```diff
@@ -0,0 +1,205 @@
+var moment = require('moment/min/moment-with-locales.min');
+var Datepicker = require('./../index');
+
+Datepicker.prototype._showView = function(view) {
+  if (this.model.get('activeView') === view) return;
+
+  this.model.set('activeView', view);
+};
+
+Datepicker.prototype._selectDate = function(momentDate, closeOnSelect) {
+  if (this._isOutOfRange(momentDate)) return;
+
+  var selectedDate = this.model.get('selectedDate');
+
+  if (!momentDate.isSame(selectedDate, 'day')) {
+    this._animateText('date', selectedDate);
+    if (!momentDate.isSame(selectedDate, 'year')) this._animateText('year', selectedDate);
+  }
+
+  this.model.set('selectedDate', momentDate);
+  this.emit('selected', this.model.get('selectedDate').format());
+
+  if (closeOnSelect) {
+    this.setValue();
+    this._hide();
+  }
+};
+
+Datepicker.prototype.action = function(action) {
+  this.emit('action', action.text || action);
+
+  if (action.disableDismiss) return;
+
+  this._hide();
+};
+
+Datepicker.prototype._selectYear = function(year) {
+  var momentDate = this.model.get('selectedDate').clone();
+  momentDate.year(year);
+
+  this._selectDate(momentDate);
+  this._setCurrentDate(momentDate);
+  
+  // REVIEW: center on selection or only on view load?
```

Comment:
It was more of a discussion when selection from list didn't change back to day-view by default, and i guess since i implemented an option to disable that functionality it could still happen... 

IMO it makes the list more jumpy if it instantly scrolls the selected item to the middle of the screen (assuming we turned on the option to stay on the list-view) but it's the way material-UI works so i felt it was better to ask. Anyways such a small difference that won't even be visible unless you opt-in so i think it could stay this way.